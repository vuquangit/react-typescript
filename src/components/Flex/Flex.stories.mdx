import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks'
import { action } from '@storybook/addon-actions'
import Flex from '.'
import Space from 'stories/components/styled-system/Space.mdx'
import Styles from 'stories/styles.mdx'

<Meta title="Story/Components/Component" component={Flex} />

<Styles />

# Flex

With `MDX` we can define a story for `Flex` right in the middle of our
Markdown documentation.

export const Template = (args) => <Flex {...args} />

<Canvas>
  <Story
    name="Normal"
  >
    {Template.bind({})}
  </Story>
</Canvas>

<div class="subheading">Props</div>

<Space />
