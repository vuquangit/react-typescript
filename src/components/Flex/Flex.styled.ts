import styled from 'styled-components'
import { space } from 'styled-system'
import { prop } from 'styled-tools'

export const FlexWrapper = styled('div')`
  ${space}

  display: flex;
  justify-content: ${prop('justifyContent', 'center')};
  align-items: ${prop('alignItems', 'center')};
  align-content: ${prop('alignContent', 'center')};
  flex-direction: ${prop('flexDirection', 'column')};
  flex-wrap: ${prop('flewWrap', 'wrap')};
  flex-grow: ${prop('flewGrow', 1)};
  order: ${prop('order', 0)};
  align-self: ${prop('alignSelf', 0)};
`
