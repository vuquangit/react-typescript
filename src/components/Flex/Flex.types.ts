import { SpaceProps } from 'styled-system'

export interface IFlexProps {
  children?: React.ReactNode
  justifyContent?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'space-evenly'
    | 'start'
    | 'end'
    | 'left'

  alignItems?:
    | 'stretch'
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'baseline'
    | 'first baseline'
    | 'last baseline'
    | 'start'
    | 'end'

  alignContent?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'space-evenly'
    | 'stretch'
    | 'start'
    | 'end'

  flexDirection?: 'row' | 'row-reverse' | 'column' | 'column-reverse'
  flexWrap?: 'nowrap' | 'wrap' | 'wrap-reverse'
  flexGrow?: number
  order?: number
  alignSelf?:
    | 'auto'
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'baseline'
    | 'stretch'
}
export declare type BaseFlexProps = IFlexProps & SpaceProps
