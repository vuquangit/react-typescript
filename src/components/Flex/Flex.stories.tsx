import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import Wrapper from '@/test/supports/AppWrapper'

import Flex from '.'
import { IFlexProps } from './Flex.types'

export default {
  title: 'Story/Components/Flex',
  component: Flex,
  argTypes: {
    flexDirection: {
      control: {
        type: 'radio',
        options: ['row', 'row-reverse', 'column', 'column-reverse'],
      },
    },
  },
} as Meta

const Template: Story<IFlexProps> = (args) => (
  <Wrapper themeMode="light">
    <Flex {...args}>
      <div>Content 1</div>
      <div>Content 2</div>
      <div>Content 3</div>
      <div>Content 4</div>
      <div>Content 5</div>
    </Flex>
  </Wrapper>
)

export const Normal = Template.bind({})
Normal.args = {
  flexDirection: 'column',
}
