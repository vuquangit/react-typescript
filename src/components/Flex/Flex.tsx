import React, { FC } from 'react'

import { FlexWrapper } from './Flex.styled'
import { BaseFlexProps } from './Flex.types'

const Flex: FC<BaseFlexProps> = ({ children }) => {
  return <FlexWrapper data-testid="Flex-wrapper">{children}</FlexWrapper>
}

export default Flex
