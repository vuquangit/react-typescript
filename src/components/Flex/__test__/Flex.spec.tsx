/**
 *
 * Tests for Flex
 *
 */

import React from 'react'
import { render } from '@testing-library/react'

import Flex from '..'
import AppWrapper from '@/test/supports/AppWrapper'

describe('App SwitchTheme', () => {
  it('Test SwitchTheme with default', async () => {
    const { container } = render(
      <AppWrapper themeMode="light">
        <Flex>
          <div>Content 1</div>
          <div>Content 2</div>
          <div>Content 3</div>
          <div>Content 4</div>
          <div>Content 5</div>
        </Flex>
      </AppWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  it('Test SwitchTheme with props', async () => {
    const { container } = render(
      <AppWrapper themeMode="light">
        <Flex
          flexDirection="column-reverse"
          justifyContent="flex-start"
          alignItems="center"
        >
          <div>Content 1</div>
          <div>Content 2</div>
          <div>Content 3</div>
          <div>Content 4</div>
          <div>Content 5</div>
        </Flex>
      </AppWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
