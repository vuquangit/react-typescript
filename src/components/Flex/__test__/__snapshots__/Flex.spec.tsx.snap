// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App SwitchTheme Test SwitchTheme with default 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: 0;
  -ms-flex-item-align: 0;
  align-self: 0;
}

<div>
  <div
    class="c0"
    data-testid="Flex-wrapper"
  >
    <div>
      Content 1
    </div>
    <div>
      Content 2
    </div>
    <div>
      Content 3
    </div>
    <div>
      Content 4
    </div>
    <div>
      Content 5
    </div>
  </div>
</div>
`;

exports[`App SwitchTheme Test SwitchTheme with props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: 0;
  -ms-flex-item-align: 0;
  align-self: 0;
}

<div>
  <div
    class="c0"
    data-testid="Flex-wrapper"
  >
    <div>
      Content 1
    </div>
    <div>
      Content 2
    </div>
    <div>
      Content 3
    </div>
    <div>
      Content 4
    </div>
    <div>
      Content 5
    </div>
  </div>
</div>
`;
